bug: Non-Deterministic Test JoinGroupAuthzIT.shouldEnforceAccessToTopics (COORDINATOR_LOAD_IN_PROGRESS)

A test that was previously fixed in #3884 is failing again with a different timing issue.

## Failed Test

`io.kroxylicious.it.filter.authorization.JoinGroupAuthzIT.shouldEnforceAccessToTopics(VersionSpecificVerification)[1]`

https://github.com/kroxylicious/kroxylicious/actions/runs/25730619539/job/75555729903?pr=3933

## Stack Trace

```
[ERROR] Tests run: 20, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 214.0 s <<< FAILURE! -- in io.kroxylicious.it.filter.authorization.JoinGroupAuthzIT
[ERROR] io.kroxylicious.it.filter.authorization.JoinGroupAuthzIT.shouldEnforceAccessToTopics(VersionSpecificVerification)[1] -- Time elapsed: 1.430 s <<< FAILURE!
org.opentest4j.AssertionFailedError: 

expected: NONE
 but was: COORDINATOR_LOAD_IN_PROGRESS
	at io.kroxylicious.it.filter.authorization.JoinGroupAuthzIT$JoinGroupEquivalence.assertUnproxiedResponses(JoinGroupAuthzIT.java:180)
	at io.kroxylicious.it.filter.authorization.AuthzIT.verifyApiEquivalence(AuthzIT.java:562)
	at io.kroxylicious.it.filter.authorization.AuthzIT$Equivalence.verifyBehaviour(AuthzIT.java:247)
	at io.kroxylicious.it.filter.authorization.JoinGroupAuthzIT.shouldEnforceAccessToTopics(JoinGroupAuthzIT.java:220)
```

## Analysis

This is a recurrence of the flaky test originally reported in #3739 and fixed by #3884. However, the error code is different:

- **Original issue (#3739)**: Error code 16 (`GROUP_COORDINATOR_NOT_AVAILABLE`)
- **Previous fix (#3884)**: Added retry on `NOT_COORDINATOR` error
- **This failure**: Error code 14 (`COORDINATOR_LOAD_IN_PROGRESS`)

The test is still experiencing timing/race conditions related to coordinator readiness, but now manifesting as `COORDINATOR_LOAD_IN_PROGRESS` rather than the previously addressed coordinator errors. This suggests the coordinator is found but not yet ready to handle requests.

## Related Issues

- #3739 - Original flaky test report
- #3884 - Previous fix (insufficient)

--- a/kroxylicious-integration-tests/src/test/java/io/kroxylicious/it/filter/authorization/AuthzIT.java
+++ b/kroxylicious-integration-tests/src/test/java/io/kroxylicious/it/filter/authorization/AuthzIT.java
@@ -619,4 +619,9 @@ public abstract class AuthzIT extends BaseIT {
         }
     }
 
+    static boolean isTransientCoordinatorError(short errorCode) {
+        Errors error = Errors.forCode(errorCode);
+        return error == Errors.NOT_COORDINATOR || error == Errors.COORDINATOR_LOAD_IN_PROGRESS || error == Errors.COORDINATOR_NOT_AVAILABLE;
+    }
+
 }

--- a/kroxylicious-integration-tests/src/test/java/io/kroxylicious/it/filter/authorization/HeartbeatAuthzIT.java
+++ b/kroxylicious-integration-tests/src/test/java/io/kroxylicious/it/filter/authorization/HeartbeatAuthzIT.java
@@ -153,6 +153,14 @@ public class HeartbeatAuthzIT extends AuthzIT {
         }
 
         @Override
+        public boolean needsRetry(HeartbeatResponseData response) {
+            Errors error = Errors.forCode(response.errorCode());
+            return error == Errors.NOT_COORDINATOR
+                    || error == Errors.COORDINATOR_LOAD_IN_PROGRESS
+                    || error == Errors.COORDINATOR_NOT_AVAILABLE;
+        }
+
+        @Override
         public void assertUnproxiedResponses(Map<String, HeartbeatResponseData> unproxiedResponsesByUser) {
             Errors aliceError = Errors.forCode(unproxiedResponsesByUser.get(ALICE).errorCode());
             Errors bobError = Errors.forCode(unproxiedResponsesByUser.get(BOB).errorCode());

--- a/kroxylicious-integration-tests/src/test/java/io/kroxylicious/it/filter/authorization/HeartbeatAuthzIT.java
+++ b/kroxylicious-integration-tests/src/test/java/io/kroxylicious/it/filter/authorization/HeartbeatAuthzIT.java
@@ -154,10 +154,7 @@ public class HeartbeatAuthzIT extends AuthzIT {
 
         @Override
         public boolean needsRetry(HeartbeatResponseData response) {
-            Errors error = Errors.forCode(response.errorCode());
-            return error == Errors.NOT_COORDINATOR
-                    || error == Errors.COORDINATOR_LOAD_IN_PROGRESS
-                    || error == Errors.COORDINATOR_NOT_AVAILABLE;
+            return isTransientCoordinatorError(response.errorCode());
         }
 
         @Override

--- a/kroxylicious-integration-tests/src/test/java/io/kroxylicious/it/filter/authorization/JoinGroupAuthzIT.java
+++ b/kroxylicious-integration-tests/src/test/java/io/kroxylicious/it/filter/authorization/JoinGroupAuthzIT.java
@@ -156,7 +156,10 @@ public class JoinGroupAuthzIT extends AuthzIT {
 
         @Override
         public boolean needsRetry(JoinGroupResponseData response) {
-            return Errors.forCode(response.errorCode()) == Errors.NOT_COORDINATOR;
+            Errors error = Errors.forCode(response.errorCode());
+            return error == Errors.NOT_COORDINATOR
+                    || error == Errors.COORDINATOR_LOAD_IN_PROGRESS
+                    || error == Errors.COORDINATOR_NOT_AVAILABLE;
         }
 
         @Override

--- a/kroxylicious-integration-tests/src/test/java/io/kroxylicious/it/filter/authorization/JoinGroupAuthzIT.java
+++ b/kroxylicious-integration-tests/src/test/java/io/kroxylicious/it/filter/authorization/JoinGroupAuthzIT.java
@@ -156,10 +156,7 @@ public class JoinGroupAuthzIT extends AuthzIT {
 
         @Override
         public boolean needsRetry(JoinGroupResponseData response) {
-            Errors error = Errors.forCode(response.errorCode());
-            return error == Errors.NOT_COORDINATOR
-                    || error == Errors.COORDINATOR_LOAD_IN_PROGRESS
-                    || error == Errors.COORDINATOR_NOT_AVAILABLE;
+            return isTransientCoordinatorError(response.errorCode());
         }
 
         @Override

--- a/kroxylicious-integration-tests/src/test/java/io/kroxylicious/it/filter/authorization/LeaveGroupAuthzIT.java
+++ b/kroxylicious-integration-tests/src/test/java/io/kroxylicious/it/filter/authorization/LeaveGroupAuthzIT.java
@@ -162,6 +162,14 @@ class LeaveGroupAuthzIT extends AuthzIT {
         }
 
         @Override
+        public boolean needsRetry(LeaveGroupResponseData response) {
+            Errors error = Errors.forCode(response.errorCode());
+            return error == Errors.NOT_COORDINATOR
+                    || error == Errors.COORDINATOR_LOAD_IN_PROGRESS
+                    || error == Errors.COORDINATOR_NOT_AVAILABLE;
+        }
+
+        @Override
         public LeaveGroupRequestData requestData(String user, BaseClusterFixture clusterFixture) {
             LeaveGroupRequestData request = new LeaveGroupRequestData();
             request.setGroupId(group);

--- a/kroxylicious-integration-tests/src/test/java/io/kroxylicious/it/filter/authorization/LeaveGroupAuthzIT.java
+++ b/kroxylicious-integration-tests/src/test/java/io/kroxylicious/it/filter/authorization/LeaveGroupAuthzIT.java
@@ -163,10 +163,7 @@ class LeaveGroupAuthzIT extends AuthzIT {
 
         @Override
         public boolean needsRetry(LeaveGroupResponseData response) {
-            Errors error = Errors.forCode(response.errorCode());
-            return error == Errors.NOT_COORDINATOR
-                    || error == Errors.COORDINATOR_LOAD_IN_PROGRESS
-                    || error == Errors.COORDINATOR_NOT_AVAILABLE;
+            return isTransientCoordinatorError(response.errorCode());
         }
 
         @Override

--- a/kroxylicious-integration-tests/src/test/java/io/kroxylicious/it/filter/authorization/SyncGroupAuthzIT.java
+++ b/kroxylicious-integration-tests/src/test/java/io/kroxylicious/it/filter/authorization/SyncGroupAuthzIT.java
@@ -160,6 +160,14 @@ class SyncGroupAuthzIT extends AuthzIT {
         }
 
         @Override
+        public boolean needsRetry(SyncGroupResponseData response) {
+            Errors error = Errors.forCode(response.errorCode());
+            return error == Errors.NOT_COORDINATOR
+                    || error == Errors.COORDINATOR_LOAD_IN_PROGRESS
+                    || error == Errors.COORDINATOR_NOT_AVAILABLE;
+        }
+
+        @Override
         public SyncGroupRequestData requestData(String user, BaseClusterFixture clusterFixture) {
             SyncGroupRequestData request = new SyncGroupRequestData();
             request.setGroupId(group);

--- a/kroxylicious-integration-tests/src/test/java/io/kroxylicious/it/filter/authorization/SyncGroupAuthzIT.java
+++ b/kroxylicious-integration-tests/src/test/java/io/kroxylicious/it/filter/authorization/SyncGroupAuthzIT.java
@@ -161,10 +161,7 @@ class SyncGroupAuthzIT extends AuthzIT {
 
         @Override
         public boolean needsRetry(SyncGroupResponseData response) {
-            Errors error = Errors.forCode(response.errorCode());
-            return error == Errors.NOT_COORDINATOR
-                    || error == Errors.COORDINATOR_LOAD_IN_PROGRESS
-                    || error == Errors.COORDINATOR_NOT_AVAILABLE;
+            return isTransientCoordinatorError(response.errorCode());
         }
 
         @Override

